core/src/test/java/org/elasticsearch/discovery/DiscoveryWithServiceDisruptionsIT.java

```diff
@@ -883,6 +888,74 @@ public void sendRequest(DiscoveryNode node, long requestId, String action, Trans
         internalCluster().stopRandomNonMasterNode();
     }
 
+    // simulate handling of sending shard failure to an isolated master
+    public void testSendingShardFailureToPartitionedMaster() throws Exception {
+        List<String> nodes = startCluster(3, 3);
```

Comment:
can we test this with a master swap as well? this will require 2 min_master_nodes..